### PR TITLE
Documentation Updates

### PR DESCRIPTION
**Pull request by HTMLGhozt**
Fixes issues in #20.

## [can-value.html](https://www.canjs.com/doc/can-value.html)
- [x] indentation
- [x] codepen
  - [x] Since most of the examples were split to provide commentary I added an examples folder in the doc folder and used `@sourceref` to link the examples.
#### discourse:
- [x] Should [this](https://github.com/canjs/can-value/blob/master/can-value.md) file be moved into the `doc` folder?

## [can-value.bind.html](https://www.canjs.com/doc/can-value.bind.html)
- [x] indentation
- [x] codepen
  - [x] Example was verbatim to the [can-value bind example](https://www.canjs.com/doc/can-value.html#Bindtootherobjects) thus it can have the same `@sourceref`.

## [can-value.from.html](https://www.canjs.com/doc/can-value.from.html)
- [x] indentation
- [x] codepen

## [can-value.returnedBy.html](https://www.canjs.com/doc/can-value.returnedBy.html)
- [x] indentation
- [x] codepen
  - [x] Example was verbatim to the [can-value returnBy example](https://www.canjs.com/doc/can-value.html#Observablederivedfromothervalues) thus it can have the same `@sourceref`.

## [can-value.to.html](https://www.canjs.com/doc/can-value.to.html)
- [x] indentation
- [x] codepen

## [can-value.with.html](https://www.canjs.com/doc/can-value.with.html)
- [x] indentation
- [x] codepen

### CODE REVIEW

File: can-value.md [hunk 1]
```diff
@@ -38,61 +38,37 @@
...
 create an observable from another initial value:
 
-```js
-import value from "can-value";
-
-const observable = value.with(15);
-
-observable.value; // is 15
-```
+@sourceref ./doc/examples/observableNumber.js
```
Comment (justinbmeyer) [inline can-value.md]: why the sourceref?  This makes it a bit harder to edit.

File: can-value.md [hunk 2]
```diff
@@ -38,61 +38,37 @@
...
+@codepen
+@highlight 3,only
```
Comment (justinbmeyer) [inline can-value.md]: I would probably not include `,only` here.  It's good to see the full example so people get an idea how it works.

File: can-value.md [hunk 3]
```diff
@@ -38,61 +38,37 @@
 At its simplest, [can-value.with] can be used to
 create an observable from another initial value:
```
Comment (justinbmeyer) [inline can-value.md]: We should also include some introduction to the example.  Examples should try as much as possible to have real-world names/ideas.  For example, 

> The following shows creating an `age` observable, reading it's value, and then listening to when the page changes:

```js
import {value, Reflect as canReflect} from "can";

const age = value.with(15);

console.log( age.value ); //-> 15

canReflect.onValue(age, function(newValue) {
  console.log( newValue ); //-> 16 ... time to drive!
});

age.value = 16;
```

File: can-value.md [hunk 4]
```diff
@@ -38,61 +38,37 @@
...
-
-const observable = value.with(15);
-
-observable.value; // is 15
-```
+@sourceref ./doc/examples/observableNumber.js
+@codepen
+@highlight 3,only
 
 You can use [can-reflect/observe.onValue] to listen for
```
Comment (justinbmeyer) [inline can-value.md]: I'd probably remove this and say something like:

"Notice that `canReflect.onValue()` can be used to listen for changes"

File: can-value.md [hunk 5]
```diff
@@ -38,61 +38,37 @@
...
-	return person.first + " " + person.last;
-} );
-
-fullName.value; // is "Grace Murray"
-```
+@sourceref ./doc/examples/observableFullName.js
+@codepen
+@highlight 5-7,only
 
 You can use [can-reflect/observe.onValue] to listen for
```
Comment (justinbmeyer) [inline can-value.md]: yeah, I wouldn't repeat this over and over ...

File: can-value.md [hunk 6]
```diff
@@ -101,31 +77,21 @@ Use `can-value` when you need an observable that can get or set a property on an
 In the example below, we use [can-value.bind] to get an observable that
 can get _and_ set `outer.inner.key`:
```
Comment (justinbmeyer) [inline can-value.md]: does this actually work?  What if you set `outer.inner = { key: "abc" }`?

File: can-value.md [hunk 7]
```diff
@@ -101,31 +77,21 @@ Use `can-value` when you need an observable that can get or set a property on an
...
 
 We can also set `keyObservable.value` to change the value at `outer.inner.key`:
 
-```js
-keyObservable.value = "aloha";
-// Now outer.inner.key === "aloha"
-```
+@sourceref ./doc/examples/observableObject.js
+@codepen
+@highlight 12,only
```
Comment (justinbmeyer) [inline can-value.md]: This page seems to be missing why someone would use `.from` vs `.to` vs `.bind` ...

Typically these are used with dynamic components.  We should make sure this is known.  You might not be able to write this up, but create an issue about it for it to be done later.

File: doc/bind.md [hunk 8]
```diff
@@ -5,26 +5,12 @@
 
 @signature `value.bind( object, keyPath )`
 
-In the example below, a `keyObservable` is created that is two-way bound to the
```
Comment (justinbmeyer) [inline doc/bind.md]: probably should be a `KeyObservable` and we should probably eventually create docs for it.

File: doc/examples/observableFullName.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import {DefineMap, value, Reflect as canReflect} from "can";
+
+const person = new DefineMap( { first: "Grace", last: "Murray" } );
```
Comment (justinbmeyer) [inline doc/examples/observableFullName.js]: I wonder if we should show this also without define map:

```
var first =value.with("Grace"),
    last = value.with("Murray");

var fullName = value.returnedBy(function(){
  return first.value + last.value
});
````

File: doc/with.md [hunk 10]
```diff
@@ -5,33 +5,34 @@
 
 @signature `value.with( initialValue )`
 
-Creates an observable value that can be read, written, and observed using [can-reflect].
+  Creates an observable value that can be read, written, and observed using [can-reflect].
```
Comment (justinbmeyer) [inline doc/with.md]: I don't think we really should show using `canReflect` unless we need to ...  folks should use `.value` to read and write.

File: can-value.md (hunk 6 above)
Comment (HTMLGhozt) [reply can-value.md]: It works as expected if you set `outer.inner = { key: "abc" };` (a matching object). It returns an error if `outer.inner` isn't a matching object, like setting it to the number `3` it throws an error. `outer.inner = 3;` errors out when trying to set the `keyObservable` -> `Uncaught TypeError: Cannot create property 'key' on number '3'`

Demo: https://codepen.io/TDillard/pen/VEVGOz?editors=0011

File: doc/bind.md (hunk 8 above)
Comment (HTMLGhozt) [reply doc/bind.md]: I'm assuming that this should be an issue?

File: can-value.md (hunk 7 above)
Comment (HTMLGhozt) [reply can-value.md]: Opened #22

**Pull request merged**